pubtools/pulplib/_impl/client/client.py

```diff
@@ -181,10 +184,52 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def get_maintenance_report(self):
+        """Get maintenance report from maintenance repository
+
+        Returns:
+            Future[:class:`~pubtools.pulplib.MaintenceReport`]
+                A future describes the maintenance status
+        """
+        report_ft = self._do_get_maintenance()
+
+        return f_map(
+            report_ft,
+            lambda data: MaintenanceReport.from_data(data)
+            if data
+            else MaintenanceReport(),
+        )
+
+    def set_maintenance(self, report):
+        """Set repositories to maintenance mode.
+
+        Args:
+            report:
+                An updated :class:`~pubtools.pulplib.MaintenanceReport` object that
+                will be used as the newest maintenance report.
+        Return:
+            Future[:class:`~pubtools.pulplib.Task`]
```

Comment:
I think this is not accurate, is it?  Doesn't publish() return a list of tasks?

```suggestion
            Future[list[:class:`~pubtools.pulplib.Task`]]
```

Note: not 100% how to make it render in docs the best way, might need to play around.